client/my-sites/plans-grid/hooks/npm-ready/data-store/use-plans-from-types.ts

```diff
@@ -6,26 +6,36 @@ import {
 	findPlansKeys,
 	TERMS_LIST,
 	TYPE_HOSTING_TRIAL,
+	TYPE_P2_PLUS,
+	GROUP_P2,
 } from '@automattic/calypso-products';
 import warn from '@wordpress/warning';
+import type { PlansIntent } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
 
 interface Props {
 	planTypes: string[];
 	term: ( typeof TERMS_LIST )[ number ];
+	intent?: PlansIntent;
 }
 
-const usePlansFromTypes = ( { planTypes, term }: Props ): PlanSlug[] => {
+const usePlansFromTypes = ( { planTypes, term, intent }: Props ): PlanSlug[] => {
 	const plans = planTypes.reduce( ( accum: PlanSlug[], type ) => {
 		// These plans don't have a term.
 		// We may consider to move this logic into the underlying `planMatches` function, but that would have wider implication so it's TBD
-		const planQuery = [
+		const planQuery: { type: string; term?: string; group?: string } = [
 			TYPE_FREE,
 			TYPE_ENTERPRISE_GRID_WPCOM,
 			TYPE_WOO_EXPRESS_PLUS,
 			TYPE_HOSTING_TRIAL,
+			TYPE_P2_PLUS,
 		].includes( type )
 			? { type }
 			: { type, term };
+
+		if ( intent === 'plans-p2' ) {
+			planQuery[ 'group' ] = GROUP_P2;
```

Comment:
All free plans have identical types (  `TYPE_PLAN_FREE` ) and terms. To query the correct P2 free plan, we add a `group` query property